cpp/src/io/parquet/writer_impl.cu

```diff
@@ -82,6 +82,18 @@ parquet::Compression to_parquet_compression(compression_type compression)
   }
 }
 
+/**
+ * @brief Function to calculate the memory needed to encode the column index of the given
+ * column chunk
+ */
+size_t column_index_buffer_size(gpu::EncColumnChunk* ck)
+{
+  // fixed 26 bytes + (sizeof(bool)+sizeof(int64)+2*max_statsize)*num_pages_in_chunk
+  // ck_stat_size is a good proxy for how much memory is needed for the min/max values
+  // calculating this per-chunk because the sizes can be wildly different
+  return 26 + (ck->ck_stat_size + 9) * ck->num_pages;
```

Comment:
Are the 26 and 9 values that would make sense to declare/reference (as constexpr or something) in a centralized place?

Reply:
I'll see if I can just replace the magic numbers with sizeof()'s and let the compiler optimize that away

Reply:
this seems magical even if it is sizeofs to me. I assume a stat is stored at these things. Maybe expound on the comment a bit saying more about what this is:
"Each page has a statistics value for min/max and sum. Worst case, this is storing three int64's, but it depends on the size of the actual column data type..."
Note that the above is made up, because I'm actually not sure what bytes are stored where. :)

Reply:
ok, removed the magic and added a ton of justification ;-)